node/service/src/rpc/api.rs

```diff
@@ -1,33 +1,84 @@
-use std::sync::Arc;
-
-use jsonrpc_core::{IoHandler, Result as JsonRpcResult};
-
 use client::Client;
+use jsonrpc_core::{IoHandler, Result as JsonRpcResult};
 use primitives::types::{SignedTransaction, TransactionBody, ViewCall, ViewCallResult};
+use rpc::types::SendMoneyRequest;
+use rpc::types::ViewAccountRequest;
+use std::sync::Arc;
+use rpc::types::DeployContractRequest;
+use rpc::types::CallMethodRequest;
 
 build_rpc_trait! {
     pub trait TransactionApi {
         /// Receive new transaction.
-        #[rpc(name = "receive_transaction")]
-        fn rpc_receive_transaction(&self, TransactionBody) -> JsonRpcResult<()>;
+        #[rpc(name = "send_money")]
+        fn rpc_send_money(&self, SendMoneyRequest) -> JsonRpcResult<()>;
         /// Call view function.
-        #[rpc(name = "view")]
-        fn rpc_view(&self, ViewCall) -> JsonRpcResult<ViewCallResult>;
+        #[rpc(name = "view_account")]
+        fn rpc_view_account(&self, ViewAccountRequest) -> JsonRpcResult<ViewCallResult>;
+        /// Deploy smart contract.
+        #[rpc(name = "deploy_contract")]
+        fn rpc_deploy_contract(&self, DeployContractRequest) -> JsonRpcResult<()>;
+        /// Call method on smart contract.
+        #[rpc(name = "call_method")]
+        fn rpc_call_method(&self, CallMethodRequest) -> JsonRpcResult<()>;
```

Comment:
it's really should be a transaction or at least semantics of the name should be that this will schedule a call, do consensus and then mutate state.

Reply:
not really sure it's implied that `send_money` is any different than `call_method`. and i'm not really sure API users at this level need to be concerned with internals of what is happening to accomplish this goal.

Reply:
According to @evgenykuzyakov, we need balance to be external, which means a transaction needs to both contain the amount it carries and also call some method.